File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    
    string numStr = to_string(x);
    vector<int> element;
    
    if(x <= 10)
        answer = true;
    else
    {
        for(int i=0; i<numStr.length(); ++i)
            element.push_back(numStr[i] - '0');
        int sum = 0;
        for(int i=0; i<element.size(); ++i)
            sum += element[i];
        if(x % sum == 0)
            answer = true;
        else
            answer = false;
    }
    
    return answer;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cs
using System;
using System.Linq;

public class Solution {
    public bool solution(int x) {
        bool answer = true;
        
        string str = x.ToString();
        int[] arr = new int[str.Length];
        for(int i = 0; i < str.Length; ++i)
            arr[i] = str[i] - '0';
        int sum = arr.Sum();
        if(x % sum != 0)
            answer = false;
        
        return answer;
    }
}